tests/test_writer_binary_raw.py

```diff
@@ -156,14 +171,19 @@
         (_D("1e-1"), [0x52, 0xC1, 0x01]),
         (_D("0e-1"), [0x51, 0xC1]),
         (_D("0e1"), [0x51, 0x81]),
+        (_D("-1e1"), [0x52, 0x81, 0x81]),
+        (_D("-1e0"), [0x52, 0x80, 0x81]),
+        (_D("-1e-1"), [0x52, 0xC1, 0x81]),
+        (_D("-0e-1"), [0x52, 0xC1, 0x80]),
+        (_D("-0e1"), [0x52, 0x81, 0x80]),
```

Comment:
Take a look at [Six's binary/text utilities](https://pythonhosted.org/six/#binary-and-text-data), in particular `indexbytes`, `byte2int`, and `int2byte`.